arch/src/x86_64/regs.rs

```diff
@@ -305,16 +302,25 @@ mod tests {
     }
 
     fn read_u64(gm: &GuestMemory, offset: usize) -> u64 {
-        let read_addr = GuestAddress(offset);
+        let read_addr = GuestAddress(offset as u64);
         gm.read_obj_from_addr(read_addr).unwrap()
     }
 
     fn validate_segments_and_sregs(gm: &GuestMemory, sregs: &kvm_sregs) {
-        assert_eq!(0x0, read_u64(&gm, BOOT_GDT_OFFSET));
-        assert_eq!(0xaf_9b00_0000_ffff, read_u64(&gm, BOOT_GDT_OFFSET + 8));
-        assert_eq!(0xcf_9300_0000_ffff, read_u64(&gm, BOOT_GDT_OFFSET + 16));
-        assert_eq!(0x8f_8b00_0000_ffff, read_u64(&gm, BOOT_GDT_OFFSET + 24));
-        assert_eq!(0x0, read_u64(&gm, BOOT_IDT_OFFSET));
+        assert_eq!(0x0, read_u64(&gm, BOOT_GDT_OFFSET as usize));
+        assert_eq!(
+            0xaf_9b00_0000_ffff,
+            read_u64(&gm, (BOOT_GDT_OFFSET + 8) as usize)
```

Comment:
Looking at `read_u64`, it seems its second parameter can be changed to be `u64` as well.